Scripts/RemediationScripts/Remediate-EnableImmutabilitySettingForBackupVault.ps1

```diff
@@ -0,0 +1,541 @@
+            $userInput = Read-Host -Prompt "(Y|N)"
+            if ($userInput -ne "Y") {
+                Write-Host "Immutability will not be enabled and locked on Backup Vault(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::DoubleDashLine)    
+                return
+            }
+        }
+        else {
+            Write-Host "'Force' flag is provided. Proceeding to enable and lock immutability on Backup Vault(s) without any prompts..." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+ 
+        # List for storing remediated Backup Vaults(s)
+        $backupVaultsRemediated = @()
+ 
+        # List for storing skipped Backup Vaults(s)
+        $backupVaultsSkipped = @()
+ 
+        # Enable and lock immutability on each Backup Vault
+        # Loop through the list of Backup Vault(s) which needs to be remediated.
+        $backupVaultsWithoutImmutability | ForEach-Object {
+            $backupVault = $_
+            try {
+                if (-not $_.ImmutabilityState) {
+                    Update-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -ImmutabilityState Disabled -ErrorAction Stop                   
+                }
+                $backupVaultResource = Update-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -ImmutabilityState Locked -ErrorAction Stop   
+ 
+                if ($backupVaultResource.ImmutabilityState.ToString() -ieq "LOCKED") {
+                    $backupVaultsRemediated += $backupVault
+                    $logResource = @{}    
+                    $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))    
+                    $logResource.Add("ResourceName", ($_.ResourceName))    
+                    $logRemediatedResources += $logResource    
+                }
+                else {
+                    $backupVaultsSkipped += $backupVault
+                    $logResource = @{}    
+                    $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))    
+                    $logResource.Add("ResourceName", ($_.ResourceName))
+                    $logResource.Add("Reason", "Error enabling and locking immutability for Backup Vault: [$($backupVault)]")            
+                    $logSkippedResources += $logResource    
+                }
+                 
+            }
+            catch {
+                $backupVaultsSkipped += $backupVault
+                $logResource = @{}    
+                $logResource.Add("ResourceGroupName", ($_.ResourceGroupName))    
+                $logResource.Add("ResourceName", ($_.ResourceName))    
+                $logResource.Add("Reason", "Error enabling and locking immutability for Backup Vault")        
```

Comment:
can we print a statement saying "Skipping this resource..."?

Reply:
Yes, we are providing SkippedBackupVaults.csv file too. Whatever resources get skipped will be logged in console and in csv file as well.